src/Illuminate/Foundation/Http/Middleware/Idempotent.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Illuminate\Foundation\Http\Middleware;
+
+use Closure;
+use Illuminate\Http\Request;
+use Illuminate\Cache\Repository as CacheRepository;
+use Illuminate\Contracts\Foundation\Application;
+use Illuminate\Foundation\Http\Exceptions\DuplicatedIdempotencyKeyException;
+use Illuminate\Foundation\Http\Exceptions\MissingIdempotencyKeyException;
+
+class Idempotent
+{
+    const PREFIX = "IDEMPOTENT_";
+
+    /**
+     * The application instance.
+     *
+     * @var \Illuminate\Contracts\Foundation\Application
+     */
+    protected $app;
+
+    /**
+     * The application cache repository.
+     *
+     * @var \Illuminate\Cache\Repository
+     */
+    protected $cache;
+
+    public function __construct(Application $app, CacheRepository $cache)
+    {
+        $this->app = $app;
+        $this->cache = $cache;
+    }
+
+    /**
+     * Determine if the HTTP request method is POST or PATCH & PUT.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return bool
+     */
+    protected function shouldSkip($request)
```

Comment:
i don't understand why DELETE is not skipped? Why not just "skip" GET HEAD OPTIONS and include everything else?

Reply:
the list of methods you have here doesn't matcn with that RFC draft either

Reply:
based on RFC: HTTP methods "OPTIONS", "HEAD", "GET", "PUT" and "DELETE" are idempotent while methods "POST" and "PATCH" are not. anyway i added PUT to idempotent middleware since many people use it for update requests but if we want to follow RFC i should delete skip PUT as well. tell me if we dont want support PUT as idempotent to edit my PR.